tests/unit/commands/image/test_deploy_openstack.py

```diff
@@ -0,0 +1,231 @@
+        args = self.prepare_image_deploy_command_openstack(1234)
+
+        deployment = self.get_deployment_openstack()
+        pimages = self.prepare_openstack_pimages_from_app()
+
+        prepare_mock_deploy(deployment, pimages, Appliance(), ["starting", "starting", "running"],
+                        mock_get_deployment, mock_app_get, mock_api_deploy, mock_get_deploy_status,
+                        mock_api_pimg_getall_for_app)
+
+        self.mock_openstack_retrieval(mock_app_get_account_resources)
+
+        # when
+        deploy_return = i.do_deploy(args)
+
+        # then
+        self.assertEquals(deploy_return, 0)
+
+    @patch('uforge.application.Api._Users._Pimages.Getall')
+    @patch('uforge.application.Api._Users._Deployments._Status.Getdeploystatus')
+    @patch('uforge.application.Api._Users._Scannedinstances._Scans._Images._Pimages._Deploys.Deploy')
+    @patch('uforge.application.Api._Users._Scannedinstances._Scans.Get')
+    @patch('uforge.application.Api._Users._Deployments.Get')
+    @patch('uforge.application.Api._Users._Scannedinstances._Scans._Images._Pimages._Accounts._Resources.Getaccountresources')
+    def test_do_deploy_openstack_a_scan_return_0_when_status_is_running(self, mock_scan_get_account_resources,
+                                                        mock_get_deployment, mock_scan_get, mock_scan_deploy,
+                                                        mock_get_deploy_status, prepare_mock_api_pimg_getall_for_scan):
+        # given
+        i = prepare_image()
+        args = self.prepare_image_deploy_command_openstack(1234)
+
+        deployment = self.get_deployment_openstack()
+        pimages = self.prepare_openstack_pimages_from_scan()
+
+        prepare_mock_deploy(deployment, pimages, Scan(), ["starting", "starting", "running"],
+                        mock_get_deployment, mock_scan_get, mock_scan_deploy, mock_get_deploy_status,
+                        prepare_mock_api_pimg_getall_for_scan)
+
+        self.mock_openstack_retrieval(mock_scan_get_account_resources)
+
+        # when
+        deploy_return = i.do_deploy(args)
+
+        # then
+        self.assertEquals(deploy_return, 0)
+
+    @patch('uforge.application.Api._Users._Pimages.Getall')
+    @patch('uforge.application.Api._Users._Deployments._Status.Getdeploystatus')
+    @patch('uforge.application.Api._Users._Appliances._Images._Pimages._Deploys.Deploy')
+    @patch('uforge.application.Api._Users._Appliances.Get')
+    @patch('uforge.application.Api._Users._Deployments.Get')
+    @patch('uforge.application.Api._Users._Appliances._Images._Pimages._Accounts._Resources.Getaccountresources')
+    def test_do_deploy_openstack_return_1_when_status_is_onfire(self, mock_app_get_account_resources, mock_get_deployment,
+                                                                mock_app_get, mock_api_deploy, mock_get_deploy_status,
+                                                                mock_api_pimg_getall_for_app):
```

Comment:
mock_api_publish_image_getall_for_app  ? longer but easier to read IMO

Reply:
I prefer not to change this one as the mock is for a webservices that contains pimages, not publish_image.
Can be "pimages" however but I found this variable already enough long.